lib/krikri/index_querier.rb

```diff
@@ -0,0 +1,17 @@
+module Krikri
+  # Queries the Search Index
+  class IndexQuerier
```

Comment:
Can you talk a little more about how this refactor improves things? The name feels a little confusing. Is this a query builder? How does this all fit together?